File: python/7 kyu/441.py
# https://www.codewars.com/kata/6a784d563bfd7732517d9832/train/python

# Passed

def vertical(words):
    rows = []
    if words:
        for idx in range(max(len(word) for word in words)):
            row = []
            for word in words:
                character = word[idx] if idx < len(word) else " "
                row.append(character)

            rows.append(row)
        
    result = "\n".join([" ".join(row).rstrip() for row in rows])
    return result

output = vertical(["hi", "world", "yo"])
print(output)

File: nodejs/7 kyu/441.js
// https://www.codewars.com/kata/5a2e8c0955519e54bf0000bd/train/javascript

// Passed

function checkDigit(number, index1, index2, digit) {
  let [startIdx, endIdx] = [index1, index2].sort((a, b) => a - b);
  let slicedNumber = String(number).slice(startIdx, endIdx + 1);
  let digitsToCheck = new Set(slicedNumber);
  let result = digitsToCheck.has(String(digit));
  return result;
}

const output = checkDigit(67845123654, 4, 2, 4);
console.log(output);